addons/web/static/tests/helpers/mock_server.js

```diff
@@ -1857,11 +1897,18 @@ var MockServer = Class.extend({
             var field = this.data[model].fields[field_changed];
             var value = values[field_changed];
             if (!field) {
-                console.warn("Mock: Can't write on field '" + field_changed + "' on model '" + model + "' (field is undefined)");
-                continue;
+                throw Error(`Mock: Can't write value "${JSON.stringify(value)}" on field "${field_changed}" on record "${model},${id}" (field is undefined)`);
             }
             if (_.contains(['one2many', 'many2many'], field.type)) {
                 var ids = _.clone(record[field_changed]) || [];
+
+                // fallback to command 6 when given a simple list of ids
+                if (
+                    Array.isArray(value) &&
+                    value.reduce((hasOnlyInt, val) => hasOnlyInt && Number.isInteger(val), true)
+                ) {
+                    value = [[6, 0, value]];
```

Comment:
Doesn't command 6 only make sense for many2many fields? Also, in which cases would the client send ids instead of commands? I don't like this either

Reply:
All commands work for all x2many types.

This specific conditions it also existing in Python. For x2many you can either give a list of commands, or a recordset.
When given a recordset, it acts as a replace, which is the 6 command.

Reply:
Even if the Python handles it, I still believe it's a bad idea. In which situations would the client send a list of ids instead of commands?

Reply:
All the time in this.data initial values. I didn't really feel like changing all existing tests data to transform array of ids to the same array of ids wrapped by 6 command...

Reply:
There is a difference between the way data are written in this.data, and the way data is sent by the client to the server. For instance, in this.data, we tolerate a one2many to be set to a list of ids, because it's convenient. However, the client communicates with the server by sending commands, not ids, for x2many fields. We shouldn't mix those two concerns. We can properly translate what is written in this.data to build the records, and this should have no impact on the client/server API.

Reply:
The web client itself might always use commands, but  calling`write` by RPC and sending it a list of ids for a one2many field is perfectly valid, the mock server needs to support it.